bundles/framework/publisher2/view/PanelGeneralInfo.js

```diff
@@ -1,178 +1,112 @@
-                tooltip: localization.name.tooltip,
-                value: null
+                tooltip: this.localization.name.tooltip,
+                validator: (key, value) => this.validateName(key, value),
+                value: data?.metadata?.name ? data?.metadata?.name : null
             },
             domain: {
-                label: localization.domain.label,
-                placeholder: localization.domain.placeholder,
+                label: this.localization.domain.label,
+                placeholder: this.localization.domain.placeholder,
                 helptags: 'portti,help,publisher,domain',
-                tooltip: localization.domain.tooltip,
-                value: null
+                tooltip: this.localization.domain.tooltip,
+                validator: (key, value) => this.validateDomain(key, value),
+                value: data?.metadata?.domain ? data?.metadata?.domain : null
             },
             language: {
-                value: null
+                value: data?.metadata?.language ? data?.metadata?.language : Oskari.getLang()
             }
         };
-        this.panel = null;
-    }, {
-        /**
-         * Creates the set of Oskari.userinterface.component.FormInput to be shown on the panel and
-         * sets up validation etc. Prepopulates the form fields if pData parameter is given.
-         *
-         * @method init
-         * @param {Object} pData initial data
-         */
-        init: function (pData) {
-            const me = this;
-            let selectedLang = Oskari.getLang();
+    };
 
-            me.fields.name.validator = function (value) {
-                const name = 'name';
-                const errors = [];
-                const sanitizedValue = Oskari.util.sanitize(value);
-                if (!value || !value.trim().length) {
-                    errors.push({
-                        field: name,
-                        error: me.loc.error.name
-                    });
-                    return errors;
-                }
-                if (sanitizedValue !== value) {
-                    errors.push({
-                        field: name,
-                        error: me.loc.error.nameIllegalCharacters
-                    });
-                    return errors;
-                }
-                return errors;
-            };
+    getId () {
+        return PANEL_GENERAL_INFO_ID;
+    }
 
-            me.fields.domain.validator = function (value) {
-                const name = 'domain';
-                const errors = [];
-                if (value && value.indexOf('://') !== -1) {
-                    errors.push({
-                        field: name,
-                        error: me.loc.error.domainStart
-                    });
-                    return errors;
-                }
-                return errors;
-            };
+    getLabel () {
+        return this.localization?.domain?.title || '';
```

Comment:
I think we only need to have something like getLabel() for the tool-panels. Maybe not even for them as they should know the label to render when they are created?

Reply:
Now that a step has been taken towards function components and statehandlers this is now useless.